modules/deeptools_bamCoverage.cfmod.pl

```diff
@@ -0,0 +1,128 @@
+#!/usr/bin/env perl
+use warnings;
+use strict;
+use Getopt::Long;
+use FindBin qw($Bin);
+use lib "$FindBin::Bin/../source";
+use CF::Constants;
+use CF::Helpers;
+use File::Copy qw(move);
+use POSIX;
+
+##########################################################################
+# Copyright 2014, Philip Ewels (phil.ewels@babraham.ac.uk)               #
+#                                                                        #
+# This file is part of Cluster Flow.                                     #
+#                                                                        #
+# Cluster Flow is free software: you can redistribute it and/or modify   #
+# it under the terms of the GNU General Public License as published by   #
+# the Free Software Foundation, either version 3 of the License, or      #
+# (at your option) any later version.                                    #
+#                                                                        #
+# Cluster Flow is distributed in the hope that it will be useful,        #
+# but WITHOUT ANY WARRANTY; without even the implied warranty of         #
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the          #
+# GNU General Public License for more details.                           #
+#                                                                        #
+# You should have received a copy of the GNU General Public License      #
+# along with Cluster Flow.  If not, see <http://www.gnu.org/licenses/>.  #
+##########################################################################
+
+my %requirements = (
+    'cores'         => ['2', '4'],
+    'memory'        => ['4G', '8G'],
+    'modules'       => '',
+    'time'          => sub {
+	my $cf = $_[0];
+	my $num_files = $cf->{'num_starting_merged_aligned_files'};
+	# Default to 1 if none were parsed for whatever reason
+	$num_files = ($num_files > 0) ? $num_files : 1;
+	# Ususally takes around 5 mins per file. Set to 30 mins just to be sure.
+	return CF::Helpers::minutes_to_timestamp ($num_files * 1 * 30);
+    }
+    );
+
+
+# The help text
+my $helptext = "".("-"x15)."\n DeepTools bamCoverage\n".("-"x15)."\n Takes two input files: a bam file and a _crosscorrelation.txt file, as created by phantompeaktools. Creates a bigWig coverage file, using deepTools/bamCoverage. Outputs are are bw files named basename.bw. The fragment length is set using 1) the input parameter fragmentLength 2) the file _crosscorrelation.txt and 3) default fragment length of 200 (if no input parameter is set and the file _crosscorrelation.txt doesn't exist). Currently this module expects a locally installed bamCoverage script.\n\n";
```

Comment:
I put in the line breaks, because for some reason the really long lines
messed up the indentation in emacs..

On Tuesday, June 23, 2015, Phil Ewels <notifications@github.com> wrote:
> In modules/deeptools_bamCoverage.cfmod.pl:
>
>> + 'cores' => ['2', '4'],
>> + 'memory' => ['4G', '8G'],
>> + 'modules' => '',
>> + 'time' => sub {
>> + my $cf = $_[0];
>> + my $num_files = $cf->{'num_starting_merged_aligned_files'};
>> + # Default to 1 if none were parsed for whatever reason
>> + $num_files = ($num_files > 0) ? $num_files : 1;
>> + # Ususally takes around 5 mins per file. Set to 30 mins just to be
sure.
>> + return CF::Helpers::minutes_to_timestamp ($num_files * 1 * 30);
>> + }
>> + );
>> +
>> +
>> +# The help text
>> +my $helptext = "".("-"x15)."\n DeepTools bamCoverage\n".("-"x15)."\n
Takes two input files: a bam file and a _crosscorrelation.txt file, as
created by phantompeaktools. Creates a bigWig coverage file, using
deepTools/bamCoverage. Outputs are are bw files named basename.bw. The
fragment length is set using 1) the input parameter fragmentLength 2) the
file _crosscorrelation.txt and 3) default fragment length of 200 (if no
input parameter is set and the file _crosscorrelation.txt doesn't exist).
Currently this module expects a locally installed bamCoverage script.\n\n";
>
> Line breaks are good - although some terminal programs will wrap long
lines, not all will, leading to some text being hidden. I try to stick to
lines which are (very roughly) 80 characters long.
>
> —
> Reply to this email directly or view it on GitHub.<
https://ci6.googleusercontent.com/proxy/5UxXlJ1bizFRzOpe2-C7k4EB1iKRnTgW2waNy_u3vJQHmDS2VjhpGJRCPDjwRcsSy4LE6V-5iD8JaYzoumTr2X0WsfG-9MfWYR_diIYxWAvAqcNFUBLWD9oQEV3VV2Y6YsIoe3taMSsUW0wI5BF_WEBOAX1j2Q=s0-d-e1-ft#https://github.com/notifications/beacon/AH70iUPrnOEAvpIahR-_a5lj58P25Yw7ks5oWWxfgaJpZM4FI3GN.gif
>